### PR TITLE
Add glossary to html reports

### LINKED ISSUE #269: Add legend to `html` reports
**What would you like to be added**:
Diki human-readable reports should contain a legend explaining ambiguous terminology like `Passed` and `Failed`.

### PR DESCRIPTION
**Pull request by AleksandarSavchev**
**What this PR does / why we need it**:
This PR adds glossary to all `html` reports which contains the meanings of rule check statuses.

**Which issue(s) this PR fixes**:
Fixes #269 

**Special notes for your reviewer**:

**Release note**:

```feature user
Glossary has been added to html reports that explains rule statuses.
```

### CODE REVIEW

File: pkg/rule/rule.go [hunk 1]
```diff
@@ -121,3 +121,25 @@ func GetStatusIcon(status Status) rune {
 		return '⚪'
 	}
 }
+
+// GetStatusMeaning returns the meaning of a given Status string
+func GetStatusMeaning(status Status) string {
```
Comment (dimityrmirchev) [inline pkg/rule/rule.go:126]:
```suggestion
func StatusDescription(status Status) string {
```

Comment (dimityrmirchev) [reply pkg/rule/rule.go:126]: Let's rename meaning to description everywhere

File: pkg/rule/rule.go [hunk 2]
```diff
@@ -121,3 +121,25 @@ func GetStatusIcon(status Status) rune {
 		return '⚪'
 	}
 }
+
+// GetStatusMeaning returns the meaning of a given Status string
```
Comment (dimityrmirchev) [inline pkg/rule/rule.go:125]:
```suggestion
// GetStatusMeaning returns the meaning of a given [Status] string.
```

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Can you please upload an image of how this is going to look in the report?

**Comment by AleksandarSavchev** [comment]
> Can you please upload an image of how this is going to look in the report?

image


image

File: pkg/rule/rule.go [hunk 3]
```diff
@@ -121,3 +121,25 @@ func GetStatusIcon(status Status) rune {
...
+func StatusDescription(status Status) string {
+	switch status {
+	case Passed:
+		return "Ran rule check has been Fulfilled."
+	case Failed:
+		return "Ran rule check has been Unfulfilled, can be considered a Finding."
+	case Errored:
+		return "Ran rule check has errored during execution. It cannot be determined if the check is Fulfilled or not."
+	case Warning:
+		return "Ran rule check has encountered a condition where it cannot be determined if the check is Fulfilled or not."
```
Comment (dimityrmirchev) [inline pkg/rule/rule.go:135]:
```suggestion
		return "The rule check has encountered an ambiguous condition or configuration preventing the ability to determine if the check is fulfilled or not."
```

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**